File: java-advent-of-code/src/test/java/advent/twenty_sixteen/Day12LeonardosMonorailTest.java
package advent.twenty_sixteen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.*;

public class Day12LeonardosMonorailTest {
    @Test
    public void checkReal() {
        DailyProblem<Integer, Integer> dp = new Day12LeonardosMonorail("/twenty_sixteen/Day12-LeonardosMonorail-input.txt");

        assertEquals(318083, dp.getPart1Answer().intValue());
        assertEquals(9227737, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Integer, Integer> dp = new Day12LeonardosMonorail("/twenty_sixteen/Day12-LeonardosMonorail-example#1.txt");

        assertEquals(42, dp.getPart1Answer().intValue());
        assertEquals(42, dp.getPart2Answer().intValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_sixteen/Day12LeonardosMonorailTest.scala
package advent.twenty_sixteen

import org.scalatest.{FlatSpec, Matchers}

class Day12LeonardosMonorailTest  extends FlatSpec with Matchers {
  "2016 Day 12 - Input" should "calculate answers" in {
    val dp = new Day12LeonardosMonorail("/twenty_sixteen/Day12-LeonardosMonorail-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day12LeonardosMonorail("/twenty_sixteen/Day12-LeonardosMonorail-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}
